--- busybox-1.20.2.orig/Makefile	2013-02-20 22:28:36.455170816 +0800
+++ busybox-1.20.2/Makefile	2013-02-20 22:29:20.051168992 +0800
@@ -1012,8 +1012,8 @@ $(mrproper-dirs):
 mrproper: clean archmrproper $(mrproper-dirs)
 	$(call cmd,rmdirs)
 	$(call cmd,rmfiles)
-	@find . -name Config.src | sed 's/.src$$/.in/' | xargs -r rm -f
-	@find . -name Kbuild.src | sed 's/.src$$//' | xargs -r rm -f
+	@find . -name Config.src | sed 's/.src$$/.in/' | xargs rm -f
+	@find . -name Kbuild.src | sed 's/.src$$//' | xargs rm -f
 
 # distclean
 #
