--- glibc-2.16.0.orig/Makeconfig.orig	2012-07-15 14:49:48.296160375 +0800
+++ glibc-2.16.0/Makeconfig	2012-07-15 15:06:20.837082124 +0800
@@ -521,9 +521,9 @@ ifneq ($(have-cc-with-libunwind),yes)
 else
   libunwind = -lunwind
 endif
-libgcc_eh := -Wl,--as-needed -lgcc_s $(libunwind) -Wl,--no-as-needed
+libgcc_eh := -Wl,--as-needed $(libunwind) -Wl,--no-as-needed
 gnulib := -lgcc $(libgcc_eh)
-static-gnulib := -lgcc -lgcc_eh $(libunwind)
+static-gnulib := -lgcc  $(libunwind)
 libc.so-gnulib := -lgcc
 endif
 +preinit = $(addprefix $(csu-objpfx),crti.o)
