diff -uNpr linux-3.7.4.orig/scripts/mod/elf.h linux-3.7.4/scripts/mod/elf.h
--- linux-3.7.4.orig/scripts/mod/elf.h	1970-01-01 05:00:00.000000000 +0500
+++ linux-3.7.4/scripts/mod/elf.h	2013-02-03 17:17:45.195376281 +0800
@@ -0,0 +1,158 @@
+#include <libelf/gelf.h>
+
+/* Intel 80386 specific definitions.  */
+
+/* i386 relocs.  */
+
+#define R_386_NONE         0            /* No reloc */
+#define R_386_32           1            /* Direct 32 bit  */
+#define R_386_PC32         2            /* PC relative 32 bit */
+#define R_386_GOT32        3            /* 32 bit GOT entry */
+#define R_386_PLT32        4            /* 32 bit PLT address */
+#define R_386_COPY         5            /* Copy symbol at runtime */
+#define R_386_GLOB_DAT     6            /* Create GOT entry */
+#define R_386_JMP_SLOT     7            /* Create PLT entry */
+#define R_386_RELATIVE     8            /* Adjust by program base */
+#define R_386_GOTOFF       9            /* 32 bit offset to GOT */
+#define R_386_GOTPC        10           /* 32 bit PC relative offset to GOT */
+#define R_386_32PLT        11
+#define R_386_TLS_TPOFF    14           /* Offset in static TLS block */
+#define R_386_TLS_IE       15           /* Address of GOT entry for static TLS
+                                           block offset */
+#define R_386_TLS_GOTIE    16           /* GOT entry for static TLS block
+                                           offset */
+#define R_386_TLS_LE       17           /* Offset relative to static TLS
+                                           block */
+#define R_386_TLS_GD       18           /* Direct 32 bit for GNU version of
+                                           general dynamic thread local data */
+#define R_386_TLS_LDM      19           /* Direct 32 bit for GNU version of
+                                           local dynamic thread local data
+                                           in LE code */
+#define R_386_16           20
+#define R_386_PC16         21
+#define R_386_8            22
+#define R_386_PC8          23
+#define R_386_TLS_GD_32    24           /* Direct 32 bit for general dynamic
+                                           thread local data */
+#define R_386_TLS_GD_PUSH  25           /* Tag for pushl in GD TLS code */
+#define R_386_TLS_GD_CALL  26           /* Relocation for call to
+                                           __tls_get_addr() */
+#define R_386_TLS_GD_POP   27           /* Tag for popl in GD TLS code */
+#define R_386_TLS_LDM_32   28           /* Direct 32 bit for local dynamic
+                                           thread local data in LE code */
+#define R_386_TLS_LDM_PUSH 29           /* Tag for pushl in LDM TLS code */
+#define R_386_TLS_LDM_CALL 30           /* Relocation for call to
+                                           __tls_get_addr() in LDM code */
+#define R_386_TLS_LDM_POP  31           /* Tag for popl in LDM TLS code */
+#define R_386_TLS_LDO_32   32           /* Offset relative to TLS block */
+#define R_386_TLS_IE_32    33           /* GOT entry for negated static TLS
+                                           block offset */
+#define R_386_TLS_LE_32    34           /* Negated offset relative to static
+                                           TLS block */
+#define R_386_TLS_DTPMOD32 35           /* ID of module containing symbol */
+#define R_386_TLS_DTPOFF32 36           /* Offset in TLS block */
+#define R_386_TLS_TPOFF32  37           /* Negated offset in static TLS block */
+/* Keep this the last entry.  */
+#define R_386_NUM          38
+
+/* ARM relocs.  */
+#define R_ARM_NONE              0       /* No reloc */
+#define R_ARM_PC24              1       /* PC relative 26 bit branch */
+#define R_ARM_ABS32             2       /* Direct 32 bit  */
+#define R_ARM_REL32             3       /* PC relative 32 bit */
+#define R_ARM_PC13              4
+#define R_ARM_ABS16             5       /* Direct 16 bit */
+#define R_ARM_ABS12             6       /* Direct 12 bit */
+#define R_ARM_THM_ABS5          7
+#define R_ARM_ABS8              8       /* Direct 8 bit */
+#define R_ARM_SBREL32           9
+#define R_ARM_THM_PC22          10
+#define R_ARM_THM_PC8           11
+#define R_ARM_AMP_VCALL9        12
+#define R_ARM_SWI24             13
+#define R_ARM_THM_SWI8          14
+#define R_ARM_XPC25             15
+#define R_ARM_THM_XPC22         16
+#define R_ARM_COPY              20      /* Copy symbol at runtime */
+#define R_ARM_GLOB_DAT          21      /* Create GOT entry */
+#define R_ARM_JUMP_SLOT         22      /* Create PLT entry */
+#define R_ARM_RELATIVE          23      /* Adjust by program base */
+#define R_ARM_GOTOFF            24      /* 32 bit offset to GOT */
+#define R_ARM_GOTPC             25      /* 32 bit PC relative offset to GOT */
+#define R_ARM_GOT32             26      /* 32 bit GOT entry */
+#define R_ARM_PLT32             27      /* 32 bit PLT address */
+#define R_ARM_ALU_PCREL_7_0     32
+#define R_ARM_ALU_PCREL_15_8    33
+#define R_ARM_ALU_PCREL_23_15   34
+#define R_ARM_LDR_SBREL_11_0    35
+#define R_ARM_ALU_SBREL_19_12   36
+#define R_ARM_ALU_SBREL_27_20   37
+#define R_ARM_GNU_VTENTRY       100
+#define R_ARM_GNU_VTINHERIT     101
+#define R_ARM_THM_PC11          102     /* thumb unconditional branch */
+#define R_ARM_THM_PC9           103     /* thumb conditional branch */
+#define R_ARM_RXPC25            249
+#define R_ARM_RSBREL32          250
+#define R_ARM_THM_RPC22         251
+#define R_ARM_RREL32            252
+#define R_ARM_RABS22            253
+#define R_ARM_RPC24             254
+#define R_ARM_RBASE             255
+/* Keep this the last entry.  */
+#define R_ARM_NUM               256
+
+/* MIPS relocs.  */
+
+#define R_MIPS_NONE             0       /* No reloc */
+#define R_MIPS_16               1       /* Direct 16 bit */
+#define R_MIPS_32               2       /* Direct 32 bit */
+#define R_MIPS_REL32            3       /* PC relative 32 bit */
+#define R_MIPS_26               4       /* Direct 26 bit shifted */
+#define R_MIPS_HI16             5       /* High 16 bit */
+#define R_MIPS_LO16             6       /* Low 16 bit */
+#define R_MIPS_GPREL16          7       /* GP relative 16 bit */
+#define R_MIPS_LITERAL          8       /* 16 bit literal entry */
+#define R_MIPS_GOT16            9       /* 16 bit GOT entry */
+#define R_MIPS_PC16             10      /* PC relative 16 bit */
+#define R_MIPS_CALL16           11      /* 16 bit GOT entry for function */
+#define R_MIPS_GPREL32          12      /* GP relative 32 bit */
+
+#define R_MIPS_SHIFT5           16
+#define R_MIPS_SHIFT6           17
+#define R_MIPS_64               18
+#define R_MIPS_GOT_DISP         19
+#define R_MIPS_GOT_PAGE         20
+#define R_MIPS_GOT_OFST         21
+#define R_MIPS_GOT_HI16         22
+#define R_MIPS_GOT_LO16         23
+#define R_MIPS_SUB              24
+#define R_MIPS_INSERT_A         25
+#define R_MIPS_INSERT_B         26
+#define R_MIPS_DELETE           27
+#define R_MIPS_HIGHER           28
+#define R_MIPS_HIGHEST          29
+#define R_MIPS_CALL_HI16        30
+#define R_MIPS_CALL_LO16        31
+#define R_MIPS_SCN_DISP         32
+#define R_MIPS_REL16            33
+#define R_MIPS_ADD_IMMEDIATE    34
+#define R_MIPS_PJUMP            35
+#define R_MIPS_RELGOT           36
+#define R_MIPS_JALR             37
+#define R_MIPS_TLS_DTPMOD32     38      /* Module number 32 bit */
+#define R_MIPS_TLS_DTPREL32     39      /* Module-relative offset 32 bit */
+#define R_MIPS_TLS_DTPMOD64     40      /* Module number 64 bit */
+#define R_MIPS_TLS_DTPREL64     41      /* Module-relative offset 64 bit */
+#define R_MIPS_TLS_GD           42      /* 16 bit GOT offset for GD */
+#define R_MIPS_TLS_LDM          43      /* 16 bit GOT offset for LDM */
+#define R_MIPS_TLS_DTPREL_HI16  44      /* Module-relative offset, high 16 bits */
+#define R_MIPS_TLS_DTPREL_LO16  45      /* Module-relative offset, low 16 bits */
+#define R_MIPS_TLS_GOTTPREL     46      /* 16 bit GOT offset for IE */
+#define R_MIPS_TLS_TPREL32      47      /* TP-relative offset, 32 bit */
+#define R_MIPS_TLS_TPREL64      48      /* TP-relative offset, 64 bit */
+#define R_MIPS_TLS_TPREL_HI16   49      /* TP-relative offset, high 16 bits */
+#define R_MIPS_TLS_TPREL_LO16   50      /* TP-relative offset, low 16 bits */
+#define R_MIPS_GLOB_DAT         51
+/* Keep this the last entry.  */
+#define R_MIPS_NUM              52
+
diff -uNpr linux-3.7.4.orig/scripts/mod/mk_elfconfig.c linux-3.7.4/scripts/mod/mk_elfconfig.c
--- linux-3.7.4.orig/scripts/mod/mk_elfconfig.c	2013-02-03 17:14:43.071383898 +0800
+++ linux-3.7.4/scripts/mod/mk_elfconfig.c	2013-02-03 17:18:05.735375422 +0800
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <elf.h>
+#include "elf.h"
 
 int
 main(int argc, char **argv)
diff -uNpr linux-3.7.4.orig/scripts/mod/modpost.h linux-3.7.4/scripts/mod/modpost.h
--- linux-3.7.4.orig/scripts/mod/modpost.h	2013-02-03 17:14:43.071383898 +0800
+++ linux-3.7.4/scripts/mod/modpost.h	2013-02-03 17:18:49.383373597 +0800
@@ -7,7 +7,7 @@
 #include <sys/mman.h>
 #include <fcntl.h>
 #include <unistd.h>
-#include <elf.h>
+#include "elf.h"
 
 #include "elfconfig.h"
 
