--- Python-3.3.0.orig/setup.py	2012-09-29 16:00:50.000000000 +0800
+++ Python-3.3.0/setup.py	2013-01-04 14:57:18.787882389 +0800
@@ -430,8 +430,6 @@ class PyBuildExt(build_ext):
         ret = os.system('%s -E -v - </dev/null 2>%s 1>/dev/null' % (gcc, tmpfile))
         is_gcc = False
         in_incdirs = False
-        inc_dirs = []
-        lib_dirs = []
         try:
             if ret >> 8 == 0:
                 with open(tmpfile) as fp:
@@ -458,12 +456,15 @@ class PyBuildExt(build_ext):
         # Ensure that /usr/local is always used, but the local build
         # directories (i.e. '.' and 'Include') must be first.  See issue
         # 10520.
+        inc_dirs = []
+        lib_dirs = []
         if not cross_compiling:
             add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
         if cross_compiling:
-            self.add_gcc_paths()
+            add_dir_to_list(self.compiler.library_dirs, '${SYSROOT}/usr/lib64')
+            add_dir_to_list(self.compiler.include_dirs, '${SYSROOT}/usr/include')
         self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
