diff -uNpr qt-creator-2.5.2-src-orig/src/plugins/projectexplorer/projectexplorer.cpp qt-creator-2.5.2-src/src/plugins/projectexplorer/projectexplorer.cpp
--- qt-creator-2.5.2-src-orig/src/plugins/projectexplorer/projectexplorer.cpp	2012-08-08 21:47:06.000000000 +0800
+++ qt-creator-2.5.2-src/src/plugins/projectexplorer/projectexplorer.cpp	2012-11-13 10:42:40.295471767 +0800
@@ -950,7 +950,7 @@ bool ProjectExplorerPlugin::initialize(c
         d->m_projectExplorerSettings.useJom =
                 s->value(QLatin1String("ProjectExplorer/Settings/UseJom"), true).toBool();
         d->m_projectExplorerSettings.autorestoreLastSession =
-                s->value(QLatin1String("ProjectExplorer/Settings/AutoRestoreLastSession"), false).toBool();
+                s->value(QLatin1String("ProjectExplorer/Settings/AutoRestoreLastSession"), true).toBool();
         d->m_projectExplorerSettings.prompToStopRunControl =
                 s->value(QLatin1String("ProjectExplorer/Settings/PromptToStopRunControl"), false).toBool();
         d->m_projectExplorerSettings.maxAppOutputLines =
